### PR TITLE
Use database to log in and validate token

### PR DESCRIPTION
**Pull request by gberaudo**

### CODE REVIEW

**Comment by gberaudo** [comment]
This PR is now ready for review.

**Comment by tsauerwein** [comment]
Good work! :+1:

**Comment by gberaudo** [comment]
I updated the PR. Now all tests must run with authentication/authorization enabled.
All post/put requests were updated to:
- first check that a 403 response is received if no credentials are sent;
- then that the test pass with the credentials of the `contributor` user.

Auth is still disabled by default in the ini file so that we have time to change the ui.

**Comment by tsauerwein** [comment]
Thanks for updating the tests!

**Comment by gberaudo** [comment]
Added index on token expire column.

**Comment by gberaudo** [comment]
I just tested creating a waypoint using the UI and it works as expected (no authorization checks).
So I think that when Travis passes it will be ready for merging in master.

**Comment by gberaudo** [comment]
Passed. Everyone OK for the merging?

**Comment by asaunier** [comment]
No objection from my perspective.

**Comment by tsauerwein** [comment]
Yes, please merge.

**Comment by gberaudo** [comment]
Thanks @asaunier and @tsauerwein for the review.

File: Makefile [hunk 1]
```diff
@@ -43,7 +43,8 @@ cleanall: clean
 
 .PHONY: test
 test: .build/venv/bin/nosetests test.ini
-	.build/venv/bin/nosetests
+	# All tests are now run with authentication/authorization enabled
```
Comment (tsauerwein) [inline Makefile]: Now that all tests are run with authentication enabled, do we still need the flag `FORCE_AUTHORIZATION_ENABLED`?

Comment (gberaudo) [inline Makefile]: Right, see https://github.com/c2corg/v6_api/pull/84

**Pull request merged**